src/retsnoop.c

```diff
@@ -275,6 +282,8 @@ static error_t parse_arg(int key, char *arg, struct argp_state *state)
 			err = append_str_file(&env.allow_globs, &env.allow_glob_cnt, arg + 1);
 			if (err)
 				return err;
+		} else if (arg[0] == ':') {
+			err = append_compile_unit(&env.allow_globs, &env.allow_glob_cnt, arg + 1);
```

Comment:
you are ignoring error here. Let's refactor these if/else chains into something like:

```
if (arg[0] == '@')
    err = append_str_file(...);
else if (args[0] == ':')
    err = append_compile_unit(...);
else
    err = append_str(...)
if (err)
    return err;
```

Same in other places